File: 429/Mukomashina Adelya/[1].cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Globalization;
using System.Text;
using System.Xml.Serialization;
//кортеж с личной информацией
(string, int, int, int) person = ("Mukomashina Adelya Raisovna", 3, 12, 2003);

// словарь-аттестат
var attestat = new Dictionary<string, int>()
{
    {"Русский язык", 5},
    {"Литература", 4},
    {"Английский язык", 4},
    {"Немецкий язык", 4},
    {"Алгебра", 4},
    {"Геометрия", 4},
    {"Информатика", 5},
    {"История России", 4},
    {"Всеобщая история", 5},
    {"Обществознание", 4},
    {"География", 4},
    {"Биология", 4},
    {"Физика", 4},
    {"Химия", 4},
    {"Музыка", 5},
};

//список родственников
List<string> rodnya = new List<string>() { "Alsu", "Rais", "Roziya", "Anver", "Kamilya", "Rustam", "Alsu" };

// строка с именем моей пушистой кивой
string kiva = "Abrikosik";
double summ = 0;
int item = 0;

//1
foreach (var ball in attestat)
{
    summ += ball.Value;
    item += 1;
}

Console.WriteLine($"Средняя оценка в аттестате: {summ / item}");

//2
var unique = rodnya.Distinct().ToList();
Console.WriteLine($"Уникальные имена среди моих родственников и меня:\n {person.Item1}");
foreach (string s in unique)
    Console.WriteLine(s);
Console.WriteLine("Вывод имен родственников: ");
int dlina1 = 0;
foreach (var name in rodnya)
{
    Console.WriteLine(name);
    dlina1 += name.Length;
}

//3
int dlina = 0;
foreach (var length in attestat)
{
    dlina += length.Key.Length;
}
Console.WriteLine($"Oбщая длина всех названий предметов: {dlina}");

//4
Console.WriteLine("Уникальные буквы в названиях предметов: ");
int i = 0;
foreach (var letters in attestat)
{
    var newstr = String.Join("", letters.Key.Distinct());
    Console.WriteLine($"{i}. {letters.Key} - {newstr}");
    i++;
}

//5
StringBuilder sb = new StringBuilder(); //dynamic string
foreach (byte b in System.Text.Encoding.Default.GetBytes(kiva))
{
    sb.Append(Convert.ToString(b, 2).PadLeft(8, '0')); //padleft - слева дополняется ноликами до длины строки 8
}
Console.WriteLine($"Имя моей домашней пушистой кивы в бинарном виде: {sb}");

//6
List<string> people = new List<string>(rodnya);
people.Sort();
people.Reverse();
Console.WriteLine("Oтсортированный по алфавиту (в обратном порядке) список родственников: ");
foreach (var name in people) Console.WriteLine(name);

//7
DateTime now = DateTime.Now;
DateTime date2 = new DateTime(2003, 12, 03);
Console.WriteLine($"Количество дней и времени от моей даты рождения до текущей даты : {now.Subtract(date2).Days}");

//8
List<string> prethings = new List<string>();
foreach (var prething in attestat) prethings.Add(prething.Key);
Queue<string> things = new Queue<string>();
Console.WriteLine("Введите индекс предмета, который хотите добавить в очередь fifo(для остановки наберите на клавиатуре число >15): ");
int thing = Convert.ToInt32(Console.ReadLine());
while (thing < 15)
{
    things.Enqueue(prethings[thing]);
    thing = Convert.ToInt32(Console.ReadLine());
}
Console.WriteLine("Предметы из очереди: ");
foreach (string ii in things) Console.WriteLine(ii);

//9
string[] names_of_rulers = {"Tenoch", "Acamapochtli", "Huitzilihitl", "Chimalpopoca", "Xihuitl Temoc",
            "Itzcoatl", "Moctezuma I", "Atotoztli", "Axayacatl", "Tizoc"};
int number = (person.Item2 + Convert.ToInt32(Math.Pow(person.Item3, 2)) + person.Item4) % 21 + 1;
Console.WriteLine($"Номер ацтекского правителя: {number}, тогда это имя: {names_of_rulers[number]}");
Console.WriteLine("Введите индекс родственника от 0 до 6, имя которого вы хотите поменять: ");
int index = Convert.ToInt32(Console.ReadLine());
if (index < 0 && index > 6)
    Console.WriteLine("error!");
else
{
    people[index] = names_of_rulers[number];
    Console.WriteLine("Измененный список имен родственников: ");
    foreach (var name in people) Console.WriteLine(name);
}

//10
var relatives = new Dictionary<string, DateTime>();
string name_member_of_family;
DateTime userDateTime;
do
{
    Console.WriteLine("Введите имя родственника(или цифру для выхода): ");
    name_member_of_family = Console.ReadLine();
    if (name_member_of_family.All(char.IsLetter))
    {
        Console.WriteLine("Введите дату рождения родственника: ");
        DateTime.TryParse(Console.ReadLine(), out userDateTime);
        relatives[name_member_of_family] = userDateTime;
    }
    else
    {
        relatives = relatives.OrderBy(x => x.Value).ToDictionary(x => x.Key, x => x.Value);
        Console.WriteLine("Вы остановили ввод родственников. Словарь родственников:  ");
        foreach (var name in relatives) Console.WriteLine(name);
    }

} while (name_member_of_family.All(char.IsLetter));

//11
//Определение варианта
string[] varients = { "аликвотная последовательность", "последовательность Сильвестра", "числа трибоначчи", "числа Леонардо" };
int choice = person.Item1.Length * dlina1 % 4;
Console.WriteLine($"Мой вариант: {varients[choice]}");
Console.WriteLine("Введите число, для которого вы хотите сделать аликвотную последовательность: ");
number = Convert.ToInt32(Console.ReadLine());
var numbers = new List<int>();
Console.WriteLine($"Аликвотная последовательность для числа {number} - ");
aliquot_sequence(number);
void aliquot_sequence(int number)
{
    if (number == 0)
        return;
    int sum = 0;
    for (int i = 1; i < number; i++)
        if(number% i == 0)
            sum+= i;
    Console.WriteLine(sum);
    aliquot_sequence(sum);
}

File: 429/Suhov Alexey/[1].py
import datetime 
from collections import deque

N=("Сухов Алексей Михайлович", 5, 9, 2003)
Att={'Русский язык': 5, 'Литература': 5, 'Английский': 5, 'Алгебра': 5, 'Геометрия': 5, 'Информатика': 5, 'История России': 5, 'Всеобщая история': 5, 'Обществознание': 5, 'География': 5, 'Биология': 5, 'Физика': 5, 'Астрономия': 5, 'Химия': 5}
C=['Мария 1982', 'Михаил 1977', 'Геннадий 1941', 'Лариса 1941', 'Анастасия 2010', 'Алексей 2003', 'Иван 2003', 'Ольга 1953','Виктор 1956' ]
kiva='Жерар'

#1
s=(Att.get('Русский язык')+Att.get('Литература')+Att.get('Английский')+Att.get('Алгебра')+Att.get('Геометрия')+Att.get('Информатика')+Att.get('История России')+Att.get('Всеобщая история')+Att.get('Обществознание')+Att.get('География')+Att.get('Биология')+Att.get('Физика')+ Att.get('Астрономия')+ Att.get('Химия'))/14
print('1) Средняя оценка аттестата:', s)
print()

#2
print('2) Уникальные имена:', list(set(C)))
print()

#3
x=len('Русский язык')+len('Литература')+len('Английский')+len('Алгебра')+len('Геометрия')+len('Информатика')+len('История России')+len('Всеобщая история')+len('Обществознание')+len('География')+len('Биология')+len('Физика')+len('Астрономия')+len('Химия')
print('3) Общая длина названий предметов:', x)
print()
import datetime 
from collections import deque

#4
u = []    
for i in list(Att):                           
    u += i                              
print("4) Уникальные буквы в названиях предметов:" , set(list(u))) 
print()

#5
bk= ' '.join(format(x, 'b') for x in bytearray(kiva, 'utf-8'))
print('5) Имя кивы в бинарном виде:',bk) 
print()

#6
C.sort(reverse=True)
print('6) Отсортированный по алфавиту (в обратном порядке) список родственников:', C)
print()

#7
today = datetime.date.today()
dr = datetime.date(2003, 9, 5)
raz = abs(dr - today)
traz=raz.days
print('7) Количество дней от моей даты рождения до текущей даты:', traz)
print()

#8
k = int(input("Введите количество элементов FIFO-списке:"))
p = [] 

for j in range(1, k + 1):
    e_f = input("Введите "+ str(j)+ "-й " + " элемент списка FIFO")
    p.append(e_f)
    
print("Введённая очередь:", p)


stop = input("Введите команду остановки: ")

while True:
    index = int(input("Введите индекс элемента: "))
    element = input("Введите элемент: ")                 
    if element == stop:
        break
    p.insert(index , element)                          
    
print("8) Полученная FIFO очередь:" , p)                   
print()

#9
p = ['Tenoch', 'Acamapochtli', 'Huitzilihitl', 'Chimalpopoca', 'Xihuitl Temoc',
            'Itzcoatl', 'Moctezuma I', 'Atotoztli', 'Axayacatl', 'Tizoc', 'Ahuitzotl',
            'Moctezuma II', 'Cuitlahuac', 'Cuauhtrmoc', 'Tlacotzin', 'Motelchiuhtzin',
            'Xochiquentzin', 'Huanitzin', 'Tehuetzquititzin', 'Cecetzin', 'Cipac']

number = (5 + 9 ** 2 + 2003) % 21 + 1
print("Индекс правителя: ", number)
print()

C.sort()       
print(C)

index = int(input("Введите индекс из отсортированного списка:"))

C[index] = p[number]    
print("9) Список с замененным именем родственника:", C)
print() 

#10 
C_date=['Мария 1982', 'Михаил 1977', 'Геннадий 1941', 'Лариса 1941', 'Анастасия 2010', 'Алексей 2003', 'Иван 2003', 'Ольга 1953']
slovar={}
for i in range(len(C_date)):
    if i==len(C_date)-1:
        slovar[C_date[i]]=0
    else:
        slovar[C_date[i]]=i+1
print('10. Связный список:', slovar)
print()

number = len("Сухов Алексей Михайлович") * len (C) % 4
print(number , "= генератор аликвотной последовательности")

#11
number = int(input('Введите число любое натуральное число:'))
def ap(num):
    b = []
    apos = [num]
    while num >= 1:
        for i in range(1, num):
            if num == 1:
                aliquotPos.append(0)
            elif num % i == 0:
                b.append(i)
        summ = sum(b)
        apos.append(summ)
        num = summ
        b = []
    return apos
    
print('11. Аликвотная последовательность от вашего числа:',ap(number))
print()
